File: server/functional-test/src/test/java/org/zanata/feature/project/SetProjectVisibilityTest.java
package org.zanata.feature.project;

import org.junit.Test;
import org.zanata.feature.Trace;
import org.zanata.feature.testharness.ZanataTestCase;
import org.zanata.page.explore.ExplorePage;
import org.zanata.workflow.LoginWorkFlow;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Damian Jansen <a href="mailto:djansen@redhat.com">djansen@redhat.com</a>
 */
public class SetProjectVisibilityTest extends ZanataTestCase {

    @Trace(summary = "The administrator can delete a project")
    @Test(timeout = ZanataTestCase.MAX_SHORT_TEST_DURATION)
    public void deleteAProject() throws Exception {
        ExplorePage explore = new LoginWorkFlow()
                .signIn("admin", "admin")
                .gotoExplore()
                .searchAndGotoProjectByName("about fedora")
                .gotoSettingsTab()
                .gotoSettingsGeneral()
                .deleteProject()
                .enterProjectNameToConfirmDelete("about fedora")
                .confirmDeleteProject()
                .gotoExplore();

        assertThat(explore.getProjectSearchResults())
                .doesNotContain("about fedora")
                .as("The project is not displayed");
    }

}

File: server/functional-test/src/test/java/org/zanata/feature/project/SetProjectVisibilityTest.kt
/*
 * Copyright 2018, Red Hat, Inc. and individual contributors as indicated by the
 * @author tags. See the copyright.txt file in the distribution for a full
 * listing of individual contributors.
 *
 * This is free software; you can redistribute it and/or modify it under the
 * terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This software is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
 * site: http://www.fsf.org.
 */
package org.zanata.feature.project

import org.junit.jupiter.api.Test
import org.zanata.feature.Trace
import org.zanata.feature.testharness.ZanataTestCase
import org.zanata.workflow.LoginWorkFlow

import org.assertj.core.api.Assertions.assertThat

/**
 * @author Damian Jansen [djansen@redhat.com](mailto:djansen@redhat.com)
 */
class SetProjectVisibilityTest : ZanataTestCase() {

    @Trace(summary = "The administrator can delete a project")
    @Test
    fun deleteAProject() {
        val explore = LoginWorkFlow()
                .signIn("admin", "admin")
                .gotoExplore()
                .searchAndGotoProjectByName("about fedora")
                .gotoSettingsTab()
                .gotoSettingsGeneral()
                .deleteProject()
                .enterProjectNameToConfirmDelete("about fedora")
                .confirmDeleteProject()
                .gotoExplore()

        assertThat(explore.projectSearchResults)
                .describedAs("The project is not displayed")
                .doesNotContain("about fedora")
    }

}
